### DISCUSSION TITLE
epoching of sensor data in the experiment pipeline

### DISCUSSION THREAD

**Discussion by giadaan** [Eelbrain/Eelbrain#66] [Q&A] [answered]
Hello all,

I defined my epochs parameters in the experiment file and I am now trying to actual epoch my sensor data. When running the function 

e.make_epoch_selection(data='mag')

I get the following output:

{
	"name": "ValueError",
	"message": "'MEG0111' is not in list",
	"stack": "---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/export/home/gantonicelli/public/PROTRACK_MEG/Code/Untitled-1.ipynb Cell 5 line 1
----> <a href='vscode-notebook-cell:/export/home/gantonicelli/public/PROTRACK_MEG/Code/Untitled-1.ipynb#W4sZmlsZQ%3D%3D?line=0'>1</a> e.make_epoch_selection(data='mag')

File /opt/anaconda3/envs/eelbrain/lib/python3.11/site-packages/eelbrain/_experiment/mne_experiment.py:5004, in MneExperiment.make_epoch_selection(self, samplingrate, data, auto, overwrite, decim, **state)
   5002 else:
   5003     y_name = data
-> 5004     ds[data] = load.mne.epochs_ndvar(ds['epochs'], data=data)
   5006 if auto is not None:
   5007     if isinstance(auto, dict):

File /opt/anaconda3/envs/eelbrain/lib/python3.11/site-packages/eelbrain/_io/fiff.py:1187, in epochs_ndvar(epochs, name, data, exclude, mult, info, sensors, vmax, sysname, connectivity, proj)
   1184 if mult != 1:
   1185     x *= mult
-> 1187 sensor = sensors or sensor_dim(epochs, picks, sysname, connectivity)
   1188 time = UTS(epochs.times[0], 1. / epochs.info['sfreq'], len(epochs.times))
   1189 return NDVar(x, ('case', sensor, time), info=info_, name=name)

File /opt/anaconda3/envs/eelbrain/lib/python3.11/site-packages/eelbrain/_io/fiff.py:830, in sensor_dim(info, picks, sysname, connectivity)
    828 if names != ch_names:
    829     try:
--> 830         index = np.array([names.index(name) for name in ch_names])
    831         c_matrix = c_matrix[index][:, index]
    832     except IndexError:

File /opt/anaconda3/envs/eelbrain/lib/python3.11/site-packages/eelbrain/_io/fiff.py:830, in <listcomp>(.0)
    828 if names != ch_names:
    829     try:
--> 830         index = np.array([names.index(name) for name in ch_names])
    831         c_matrix = c_matrix[index][:, index]
    832     except IndexError:

ValueError: 'MEG0111' is not in list"
}

Any idea on how I could fix this issue? Thank you very much in advance

**Comment by proloyd** [comment]
@giadaan which eelbrain version are you using? Did you check if your files have channels with that name, manually (i.e. via opening them with `mne.io.load_raw`)? Can you try `e.load_epochs(data='mag')` and check if you can load the epochs?

**Comment by giadaan** [comment]
Hi! 

Yes, the channel is actually among my channels and no, I cannot do load.epochs (when running it, the program tells me to run make epoch selection first). 

Giada Antonicelli 
Predoctoral Researcher BCBL - Neurolinguistics and Aphasia group 
www.bcbl.eu 
Tel: +34 943 309 300 Ext #307 

Legal disclaimer/Aviso legal/Lege-oharra: www.bcbl.eu/legal-disclaimer 
We are working flexibly at BCBL. I am sending this message at a time that suits me. Please feel comfortable knowing that I do not expect you to read, respond to or action it outside of regular working hours.

**Reply by giadaan** [reply]
eelbrain version 0.39.8

**Comment by christianbrodbeck** [comment, marked as answer]
The error happens when automatically trying to load the sensor connectivity (information about which sensor is adjacent to which other sensor).

Does the issue only occur with `MneExperiment.make_epoch_selection()` or also in other contexts like `MneExperiment.load_epochs()` (use `reject=False`)? If the former, could you re-try with Eelbrain 0.40a7 (https://github.com/christianbrodbeck/Eelbrain/wiki/Installing#pre-release-versions)? Otherwise, can you share a short segment of data?

**Reply by giadaan** [reply]
If I set reject=False in load_epochs() it works! Thank you!
